src/parser/smt2/smt2_state.cpp

```diff
@@ -756,6 +775,7 @@ void Smt2State::setLogic(std::string name)
 
   // Core theory belongs to every logic
   addCoreSymbols();
+  addSkolemSymbols();
```

Comment:
Let's guard this via an option in https://github.com/cvc5/cvc5/blob/main/src/options/parser_options.toml

i.e. `parse-skolem-definitions`.